File: BossFight Rest/BossFight/models/Target.cs
using System;
using System.Linq;

namespace BossFight.Models
{
    public class Target
    {
        public int HP {get; set;}
        public string Name {get; set;}
        public int Level {get; set;}

        public Target(int pHP = 1, string pName = "No name", int pLevel = 1)
        {
            HP = pHP;
            Name = pName;
            Level = pLevel;
        }

        public virtual int GetMaxHp()
        {
            throw new NotImplementedException("get_max_hp() not implemented for class");
        }

        public virtual bool IsDead()
        {
            return HP <= 0;
        }

        public virtual bool IsAlive()
        {
            return !IsDead();
        }

        public virtual bool IsAtFullHealth()
        {
            return HP >= GetMaxHp();
        }

        public virtual object PossessiveName()
        {
            if (Name.Last() == 's')
            {
                return Name + "'";
            }
            else
            {
                return Name + "'s";
            }
        }
    }
}

File: BossFightBackEnd/boss_fight_py_edition/target.py
class Target:
    def __init__(self, hp: int = 1, name: str = "No name", level: int = 1):
        self.hp: int = hp
        self.name: str = name
        self.level: int = level

    def get_max_hp(self) -> int:
        raise NotImplementedError("get_max_hp() not implemented for class")

    def is_dead(self) -> bool:
        return self.hp <= 0

    def is_alive(self) -> bool:
        return not self.is_dead()

    def is_at_full_health(self) -> bool:
        return self.hp >= self.get_max_hp()

    def possessive_name(self) -> str:
        if self.name[-1:] == 's':
            return self.name + "'"
        else:
            return self.name + "'s"
